pkg/proxy/ipvs/proxier_test.go

```diff
@@ -2926,6 +2926,99 @@ func TestCleanLegacyService(t *testing.T) {
 
 }
 
+func TestCleanLegacyServiceWithRealServers(t *testing.T) {
+	ipt := iptablestest.NewFake()
+	ipvs := ipvstest.NewFake()
+	ipset := ipsettest.NewFake(testIPSetVersion)
+	fp := NewFakeProxier(ipt, ipvs, ipset, nil, nil)
+
+	// all deleted expect ipvs2
+	activeServices := map[string]bool{"ipvs2": true}
+	// All ipvs services in the system.
+	currentServices := map[string]*utilipvs.VirtualServer{
+		"ipvs0": { // deleted with real servers
+			Address:   net.ParseIP("1.1.1.1"),
+			Protocol:  string(v1.ProtocolUDP),
+			Port:      53,
+			Scheduler: "rr",
+			Flags:     utilipvs.FlagHashed,
+		},
+		"ipvs1": { // deleted no real server
+			Address:   net.ParseIP("2.2.2.2"),
+			Protocol:  string(v1.ProtocolUDP),
+			Port:      54,
+			Scheduler: "rr",
+			Flags:     utilipvs.FlagHashed,
+		},
+		"ipvs2": { // not deleted
+			Address:   net.ParseIP("3.3.3.3"),
+			Protocol:  string(v1.ProtocolUDP),
+			Port:      54,
+			Scheduler: "rr",
+			Flags:     utilipvs.FlagHashed,
+		},
+	}
+
+	// "ipvs0" has a real server so it shouldn't be deleted.
```

Comment:
We now delete the VS and unbind the IP when we delete the service, regardless of RealServers right?
(the code below seems to do exactly this so I assume the comment is wrong?)

Reply:
ah yeah, sorry copy/paste error, will fix